### DISCUSSION TITLE
Delete K8S Lease using Condition

### DISCUSSION THREAD

**Discussion by Wallenstein7** [fabric8io/kubernetes-client#6136] [Q&A]
Hello,

K8S spec defines for leases a way to define condition when deleting the lease.
But how can i achive it using Fabric8 ?

Thx.

**Comment by rohanKanojia** [comment]
@Wallenstein7 : Could you please elaborate on your question? Are you asking about how to delete the [Lease resource](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/lease-v1/) or add a precondition using the finalizer?

**Comment by Wallenstein7** [comment]
Hello,

i want to delete a lease by defining a condition like "if lease has resource version = X" (what currently works only for "patch"). If the condition is fulfilled, the lease would be deleted, otherwise not.

The reason for my question is the fact (as i understood), that the leases have NO TTL time which would delete them automatically (am i wrong here?)

The standard java kubernetes-client provides DeleteOptions class which can be used to setup preconditions in the context of delete operation.
